### PR TITLE
Feature/update legs and other stuff

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Current Status: Ready for Review

Done:
- Legs now being removed on a unique ID (maybe to be ail_seq_num when fully integrated)
- local state set up per leg
- legs now limited to 10 
- disabled pos num input when leg limit of 10 reached (future ticket - frequent positions gray out on limit)
- Selects now have a default empty option
- initial leg no longer grabbing from existing pos redux state
- little 0 showing under alert when no legs fixed
- send full AI info to BE

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
image

**Comment by elizabeth-jimenez** [comment]
@SidelineCory24 @scott062 Back in WIP - was going to add the create Agenda stuff in another PR, but instead going to just add it here

File: src/actions/agendaItemMaintenancePane.js [hunk 1]
```diff
@@ -29,27 +29,27 @@ export function aiCreateSuccess(data) {
...
-      // }) })
-      .get('/fsbid/reference/cycles/', { cancelToken: new CancelToken((c) => {
-        cancel = c;
-      }) })
-      .then(({ data }) => data || [])
-      .then(() => {
-        const toastTitle = UPDATE_AGENDA_ITEM_SUCCESS_TITLE;
-        const toastMessage = UPDATE_AGENDA_ITEM_SUCCESS;
+      .post('/fsbid/agenda/agenda_item/', { Data: [
+        'Elsa, Gigi, Nori, Sophie',
```
Comment (elizabeth-jimenez) [inline src/actions/agendaItemMaintenancePane.js:39]: 🐈‍⬛ 🐈‍⬛ 🐈  🐈

**Comment by elizabeth-jimenez** [comment]
Actual Save won't work until BE fully fleshed out, but to keep things moving and prevent merge conflicts, I can address that in another PR - Ticket however, will remain in Review until that new PR merges

File: src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx [hunk 2]
```diff
@@ -44,24 +48,18 @@ const AgendaItemLegsForm = props => {
...
           <Spinner type="legs" size="small" />
       }
       {
-        !numLegs &&
+        legs.length === 0 &&
         <Alert type="info" title="No Agenda Item Legs" />
       }
       {
-        !legsLoading && numLegs &&
+        !legsLoading && (legs.length > 0) &&
```
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx:62]: Wow - i'll never forgive myself. Thanks!

**Comment by SidelineCory24** [comment]
TOD, Action, and Travel are not updating when you remove a leg to the left. All other leg info is updating properly.

https://user-images.githubusercontent.com/55964163/184161115-62d7be0a-78b6-475e-813f-2a936ae45d7b.mp4

**Comment by elizabeth-jimenez** [comment]
> TOD, Action, and Travel are not updating when you remove a leg to the left. All other leg info is updating properly.
> 
>  leg_removal2.mp4

🙀 Nice find!

**Comment by SidelineCory24** [comment]
`<option selected hidden>` needs to be added to the `Status` and `Report Category` drop-downs, similar to `asgSepBid`. As of now it appears `Ready` and `Position Shootout` are selected but they don't come through the payload unless you select another option then reselect them.

File: public/config/config.json [hunk 3]
```diff
@@ -4,16 +4,16 @@
   },
   "hrAuthUrl": "http://localhost:3333/Authorize?sAppCircuitID=32",
   "flags": {
-    "static_content": true,
```
Comment (elizabeth-jimenez) [inline public/config/config.json:7]: Just alphabetized them

**Comment by SidelineCory24** [comment]
> `<option selected hidden>` needs to be added to the `Status` and `Report Category` drop-downs, similar to `asgSepBid`. As of now it appears `Ready` and `Position Shootout` are selected but they don't come through the payload unless you select another option then reselect them.

discussed offline, no change needed, working as intended for this PR

**Comment by elizabeth-jimenez** [comment]
image
it's back 😭

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx [hunk 4]
```diff
@@ -1,40 +1,48 @@
...
 
-  const userRemarks = [];
-  const [userSelections, setUserSelections] = useState(userRemarks);
-
+  const [userRemarks, setUserRemarks] = useState([]);
 
   const updateSelection = (remark) => {
-    const userSelection$ = [...userSelections];
-    const found = find(userSelection$, { seq_num: remark.seq_num });
+    const userRemarks$ = [...userRemarks];
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:32]: 🤌

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx [hunk 5]
```diff
@@ -1,40 +1,48 @@
...
-      setUserSelections(userSelection$);
+      userRemarks$.push(remark);
+      setUserRemarks(userRemarks$);
     } else {
-      setUserSelections(filter(userSelection$, (r) => r.seq_num !== remark.seq_num));
+      setUserRemarks(filter(userRemarks$, (r) => r.seq_num !== remark.seq_num));
     }
   };
 
+  const submitAI = () => {
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:42]: patent pending😎

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx [hunk 6]
```diff
@@ -73,30 +78,41 @@ const AgendaItemMaintenancePane = (props) => {
...
 
-  const saveAI = (e) => {
-    e.preventDefault();
-    const userInputs = {
+  useEffect(() => {
+    sendMaintenancePaneInfo({
       selectedPanelMLDate: selectedPanelMLDate || '',
       selectedPanelIDDAte: selectedPanelIDDate || '',
-      userSelections: userSelections || [],
+      Remarks: userRemarks || [],
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:97]: lowercase?

**Comment by elizabeth-jimenez** [comment]
https://user-images.githubusercontent.com/23065001/184257538-df9bdb9e-efde-4ecb-a98c-9f163a0b9c4c.mp4


🥲

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx (hunk 6 above)
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:97]: 🪄 done

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx (hunk 5 above)
Comment (SidelineCory24) [reply src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:42]: verified payload is sending the correct data if leg count is anywhere from 1-10

**Comment by SidelineCory24** [comment]
> Actual Save won't work until BE fully fleshed out, but to keep things moving and prevent merge conflicts, I'll address that in another PR - The JIRA ticket however, will remain in Review until that new/other PR merges. As long as all the AI info is being properly tracked and the right info hits the Action file on click of the Save AI button, then this PR is good 🤞

PR is working per this criteria, approving

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go!🚀

**Comment by SidelineCory24** [comment]
@elizabeth-jimenez you can do the honors, don't want to create the merge conflicts for you😅

**Pull request merged**